src/framework/components/script/component.js

```diff
@@ -640,6 +640,8 @@ class ScriptComponent extends Component {
         if (typeof scriptType === 'string') {
             scriptType = this.system.app.scripts.get(scriptType);
         } else if (scriptType) {
+            if (scriptType.__name === null)
+                scriptType.__name = scriptType.name;
```

Comment:
I think you can check if the name property was explicitly defined as a static member rather than inferred using `getOwnPropertyDescriptor`, which could be useful, but would need to verify. I wonder if for ES Modules, which are more likely to be minified/mangled, that we require/enforce that a `name` or similar is specifically defined 🤔